astropy/io/misc/asdf/tags/transform/compound.py

```diff
@@ -139,8 +120,7 @@ def from_tree_transform(cls, node, ctx):
                 new_mapping.append(entry)
             else:
                 new_mapping.append(i)
-                transform = transform & ConstantType.from_tree(
-                    {'value': int(entry.value)}, ctx)
+                transform = transform & Const1D(int(entry.value))
```

Comment:
Casting to int is not necessary here. I'm guessing this was the idea for the old Const model but in modeling all parameters are turned into floats.